runtime/plugin/osc52.lua

```diff
@@ -0,0 +1,26 @@
+if
+  vim.g.osc52 == false
+  or (
+    vim.g.osc52 == nil and (not os.getenv('SSH_TTY') or os.getenv('TMUX') or vim.o.clipboard ~= '')
```

Comment:
We really need a builtin way to disable plugins without special purpose variables. Often it's good enough to delete an autocmd group or unmap a mapping, but when neither of those mechanisms is in use, I guess the only way is to mess with 'runtimepath' (nasty) or to monkeypatch `vim.*` modules/functions...

In this case, `g:clipboard` may give enough control. If `g:clipboard` is set we can skip this. So let's just document that, then we don't need `g:osc52`.

Reply:
>In this case, g:clipboard may give enough control. If g:clipboard is set we can skip this. So let's just document that, then we don't need g:osc52.

How would someone opt out of using both OSC 52 _and_ `g:clipboard` in that case? For example, if they want to stick with `xsel`?

Reply:
Suggested framing: why is it any different than the other tools we look for (especially tmux, for example)?

We don't have a `g:tmuxclipboard = false` option.

If it's too cumbersome to tell the clipboard provider which tools to look for, we should fix that with something like a `vim.ui.clipboard.tools: table`, or a 'clipboardoptions' option.

In the meantime, we can document how to set g:clipboard to something "neutral".

---

P.S. For reference, I started this patch, but then realized that treating `g:clipboard=v:true` as "disable OSC52" is weird:

```diff
diff --git a/runtime/autoload/provider/clipboard.vim b/runtime/autoload/provider/clipboard.vim
index 05f6bdb871d3..b6c569a6b7c5 100644
--- a/runtime/autoload/provider/clipboard.vim
+++ b/runtime/autoload/provider/clipboard.vim
@@ -67,7 +67,10 @@ function! provider#clipboard#Error() abort
 endfunction
 
 function! provider#clipboard#Executable() abort
-  if exists('g:clipboard')
+  if get(g:, 'clipboard', v:null) is# v:false
+    let s:err = 'clipboard: disabled (g:clipboard=v:false). :help clipboard'
+    return ''
+  elseif get(g:, 'clipboard', v:true) isnot# v:true
     if type({}) isnot# type(g:clipboard)
           \ || type({}) isnot# type(get(g:clipboard, 'copy', v:null))
           \ || type({}) isnot# type(get(g:clipboard, 'paste', v:null))
```

Reply:
>Why is it any different than e.g. tmux clipboard (we don't have a g:tmuxclipboard = false option)?

This is true, but tmux doesn't have the problems that OSC 52 has: namely, the fact that OSC 52 must wait for a response from the terminal (which can take a while for some terminals, according to some users) and that some terminals simply don't allow reading from the clipboard with OSC 52 (tmux reads and writes to its own internal buffers in addition to the system clipboard, so it can sidestep these problems).

For both of these reasons I expect it will be much more common for users to want to opt out of OSC 52 than for tmux.

Really the only issue with OSC 52 is when pasting, and even then it's much easier (and better!) to use `Ctrl-V` or whatever. So this is really only an issue for people using `set clipboard=unnamed` (which is why I want to not use OSC 52 when that is set).

Reply:
> really only an issue for people using `set clipboard=unnamed` (which is why I want to not use OSC 52 when that is set).

Sure, no objection to disabling by default when clipboard=unnamed is set. 

If we absolutely need a special purpose option, as a stop-gap we could make `g:clipboard` accept a list of default provider names. So users could opt-out of osc52 by saying which tools they want to enable:

    let g:clipboard = ['xsel', 'xclip', ...]

Reply:
Sounds good, I'll remove `g:osc52` and let you do that in a follow up.